gradunwarp/core/unwarp_resample.py

```diff
@@ -411,16 +404,16 @@ def siemens_B(alpha, beta, x1, y1, z1, R0):
         f = np.power(r / R0, n)
         for m in range(0, n + 1):
             f2 = alpha[n, m] * np.cos(m * phi) + beta[n, m] * np.sin(m * phi)
-            _ptemp = utils.legendre(n, m, np.cos(theta))
-            #_ptemp = scipy.special.lpmv(m, n, np.cos(theta))
+            #_ptemp = utils.legendre(n, m, np.cos(theta))
+            _p = scipy.special.lpmv(m, n, np.cos(theta))
             normfact = 1
             # this is Siemens normalization
             if m > 0:
                 normfact = math.pow(-1, m) * \
                 math.sqrt(float((2 * n + 1) * factorial(n - m)) \
                           / float(2 * factorial(n + m)))
-            _p = normfact * _ptemp
-            b = b + f * _p * f2
+            _p *= normfact
```

Comment:
It seems cleaner to me to move `_p *= normfact` into the `if` block above, rather than unconditionally multiply.

```suggestion
                _p *= normfact
```